### PR TITLE
Metal garage gate and Mechanical winch

### PR DESCRIPTION
**Pull request by SzQ1**
#### Summary

 Features: Create entries in construction.json for metal garage gate and mechanical winch.

#### Purpose of change

Allow players to reconstruct broken mechanical winches, in the cities with wandering spawns turned on, mechanical winches often get smashed by zombies, meanwhile working garage gates are very useful and cool. 
Allow players to construct metal garage gates and mechanical winches to control them. For players that like to have cool and safe entrance at their cool base. Why player constructed metal deathmobile shouldn't have player constructed, metal garage gate to enter by.

#### Describe the solution

Created entries in construction.json for mechanical winch and garage gate.

#### Describe alternatives you've considered

Not considered.

#### Testing

Works exactly as already in game garage gates and as stated in the included "pre_note"/Annotations that player will see in construction menu. No interaction with Palisade system.

#### Additional context

-Winch only work next to (reinforced)Concrete and brick(not adobe) walls. I would like to make it work for other types of walls, don't know how yet.
-When open"lifted", "under" garage gates there always will be wooden floor. Should I add `"pre_terrain": "t_floor_noroof",` to the garage gate or include amount of wood required to build wooden floor into the components of garage gates?

Sprite-0001

### CODE REVIEW

File: data/json/construction.json [hunk 1]
```diff
@@ -4217,5 +4217,53 @@
...
+    "required_skills": [ [ "fabrication", 7 ], [ "mechanics", 2 ] ],
+    "time": "220 m",
+    "qualities": [
+      [ { "id": "HAMMER", "level": 2 } ],
+      [ { "id": "SAW_M", "level": 1 } ],
+      [ { "id": "WRENCH", "level": 1 } ],
+      [ { "id": "WELD", "level": 2 } ],
+      [ { "id": "GLARE", "level": 2 } ]
+    ],
+    "components": [
```
Comment (SzQ1) [inline data/json/construction.json:4260]: Normal metal door along with metal doorframe cost around:
`[ "steel_lump", 4 ], [ [ "steel_chunk", 16 ], [ "scrap", 48 ] ` and `[ "steel_plate", 2 ], [ "sheet_metal", 8 ]` and `[ "hinge", 3 ]`
    `"required_skills": [ [ "fabrication", 7 ] ],`
   ` "time": "120 m",`
   ` "qualities": [ [ { "id": "HAMMER", "level": 2 } ], [ { "id": "SAW_M", "level": 1 } ], [ { "id": "WRENCH", "level": 1 } ] ],`

File: data/json/construction.json [hunk 2]
```diff
@@ -4217,5 +4217,53 @@
...
+    "required_skills": [ [ "fabrication", 7 ], [ "mechanics", 3 ] ],
+    "time": "60 m",
+    "qualities": [
+      [ { "id": "HAMMER", "level": 2 } ],
+      [ { "id": "SAW_M", "level": 1 } ],
+      [ { "id": "WRENCH", "level": 1 } ],
+      [ { "id": "WELD", "level": 2 } ],
+      [ { "id": "GLARE", "level": 2 } ]
+    ],
+    "components": [
```
Comment (SzQ1) [inline data/json/construction.json:4236]: Rope and Pulley skills and cost:
`[ "fabrication", 5 ], [ "mechanics", 2 ] `
`[ [ [ "rope_makeshift_30", 1 ], [ "rope_30", 1 ], [ "vine_30", 1 ] ], [ [ "stick", 8 ], [ "2x4", 8 ] ] ],`

File: data/json/construction.json [hunk 3]
```diff
@@ -4217,5 +4217,53 @@
...
+      [ [ "steel_lump", 2 ], [ "steel_chunk", 6 ], [ "scrap", 18 ] ],
+      [ [ "chain", 1 ] ],
+      [ [ "rope_6", 2 ], [ "rope_makeshift_6", 2 ] ]
```
Comment (SzQ1) [inline data/json/construction.json:4239]: Rope and Pulley skills and cost:
`[ "fabrication", 5 ], [ "mechanics", 2 ]`
`[ [ [ "rope_makeshift_30", 1 ], [ "rope_30", 1 ], [ "vine_30", 1 ] ], [ [ "stick", 8 ], [ "2x4", 8 ] ] ],`

File: data/json/construction.json [hunk 4]
```diff
@@ -4217,5 +4217,53 @@
...
+      [ { "id": "HAMMER", "level": 2 } ],
+      [ { "id": "SAW_M", "level": 1 } ],
+      [ { "id": "WRENCH", "level": 1 } ],
+      [ { "id": "WELD", "level": 2 } ],
+      [ { "id": "GLARE", "level": 2 } ]
+    ],
+    "components": [
+      [ [ "steel_lump", 5 ], [ "steel_chunk", 15 ], [ "scrap", 45 ] ],
+      [ [ "steel_plate", 4 ], [ "sheet_metal", 16 ] ],
+      [ [ "pipe", 3 ], [ "rebar", 1 ] ]
```
Comment (SzQ1) [inline data/json/construction.json:4263]: Normal metal door along with metal doorframe cost and requirements:
`[ "steel_lump", 4 ], [ [ "steel_chunk", 16 ], [ "scrap", 48 ] and [ "steel_plate", 2 ], [ "sheet_metal", 8 ] and [ "hinge", 3 ]`
`"required_skills": [ [ "fabrication", 7 ] ],`
`"time": "120 m",`
`"qualities": [ [ { "id": "HAMMER", "level": 2 } ], [ { "id": "SAW_M", "level": 1 } ], [ { "id": "WRENCH", "level": 1 } ] ],`

**Comment by Coolthulhu** [comment]
```
(json-error)
Json error: data/mods/dda/../../json/construction.json:4 226:18: Invalid or misplaced field name "difficulty" in JSON data

    "description": "Build mechanical winch",
    "category": "CONSTRUCT",
    "difficulty":
                 ^
                  5,
    "required_skills": [ [ "fabrication", 7 ], [ "mechanics", 3 ] ],
    "time": "60 m",

20:07:53.782 ERROR DEBUGMSG : src/json.cpp:124 [void JsonObject::report_unvisited() const] (json-error)
Json error: data/mods/dda/../../json/construction.json:4 250:18: Invalid or misplaced field name "difficulty" in JSON data

    "description": "Build metal garage gate",
    "category": "CONSTRUCT",
    "difficulty":
                 ^
                  5,
    "required_skills": [ [ "fabrication", 7 ], [ "mechanics", 2 ] ],
    "time": "220 m",
```

**Comment by Coolthulhu** [comment]
The requirements aren't working correctly, it's just mechanics 3

**Comment by Coolthulhu** [comment]
There's a problem with building gates in that the player isn't adequately "briefed" on them.
The winch will not work if there's no wall of specific type between it and the door. Concrete or brick wall will work, but wood one won't. The message doesn't explain why isn't the winch working.

**Pull request merged**